go/vt/mysqlctl/s3backupstorage/s3_test.go

```diff
@@ -0,0 +1,55 @@
+package s3backupstorage
+
+import (
+	"errors"
+	"net/http"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/request"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/aws/aws-sdk-go/service/s3/s3iface"
+)
+
+type s3ErrorClient struct{ s3iface.S3API }
+
+func (s3errclient *s3ErrorClient) PutObjectRequest(in *s3.PutObjectInput) (*request.Request, *s3.PutObjectOutput) {
+	req := request.Request{
+		HTTPRequest: &http.Request{},          // without this we segfault \_(ツ)_/¯ (see https://github.com/aws/aws-sdk-go/blob/v1.28.8/aws/request/request_context.go#L13)
+		Error:       errors.New("some error"), // this forces req.Send() (which is called by the uploader) to always return non-nil error
+	}
+
+	return &req, &s3.PutObjectOutput{}
+}
+
+func TestAddFileError(t *testing.T) {
+	bh := &S3BackupHandle{client: &s3ErrorClient{}, readOnly: false}
+	wc, err := bh.AddFile(aws.BackgroundContext(), "somefile", 100000)
+
+	if err != nil {
```

Comment:
can you change all the `t.Errorf` calls to use `require.NoError`?

Reply:
of course!